tensorzero-core/src/optimization/dicl.rs

```diff
@@ -556,14 +556,20 @@ pub async fn insert_dicl_examples_with_batching(
             .iter()
             .map(|(sample, embedding)| {
                 let output = sample
-                    .output
+                    .stored_output
                     .as_ref()
-                    .and_then(|outputs| outputs.first())
-                    .map(|output| serde_json::to_string(output).unwrap_or_default())
+                    .map(|stored_output| match stored_output {
+                        StoredOutput::Chat(outputs) => {
+                            serde_json::to_string(outputs).unwrap_or_default()
+                        }
+                        StoredOutput::Json(json_output) => {
+                            serde_json::to_string(json_output).unwrap_or_default()
+                        }
+                    })
```

Comment:
let's make this a helper function on the StoredOutput type. You should be able to do #[derive(Serialize)] on StoredOutput and use `#[serde(untagged)]` so that it serializes correctly.